painttiming.bs

```diff
@@ -62,9 +87,41 @@ Formally, we consider the user agent to have "rendered" a document when it has p
 
     NOTE: The rendering pipeline is very complex, and the timestamp should be the latest timestamp the user agent is able to note in this pipeline (best effort). Typically the time at which the frame is submitted to the OS for display is recommended for this API.
 
+A [=generated content pseudo-element=] is a <dfn>paintable pseudo-element</dfn> when all of the following apply:
+* The pseudo-element's [=used=] [=visibility=] is <code>visible</code>.
+* The pseudo-element's [=used=] [=opacity=] is greater than zero.
+
+An [=element=] is <dfn export>contentful</dfn> when one or more of the following apply:
+* The element has a [=text node=] child, with at least one character, excluding [=document white space characters=].
+* The element is a [=replaced element=] representing a [=loaded=] image.
+* The element's [=used=] style contains a [=loaded=] [=background-image=].
```

Comment:
Hm... I don't think this is quite right. I don't think used style containing a background image is not a well defined concept. First off, it's used *value*, not used *style*, and I don't think a style containing a background image is a thing.

Now, reading https://drafts.csswg.org/css-backgrounds/, it appears to me that the concept & the phrase we want to be using is "the background image(s) of an element", which is a phrase repeatedly used throughout the spec.

We may also want to check the size of the background image, and make sure it's not 0px either in width or height.

Reply:
Revised.